text/0000-security-advisories.md

```diff
@@ -0,0 +1,297 @@
+in the example in [Motivation](#motivation), don't yet have a good reason to do
+so. A security issue would be a good reason, but marking a package as
+deprecated does not imply that.
+
+## Extending yanking for security advisories
+
+It has been proposed to [extend the semantics of yanking such that it could be
+used for security advisories](https://github.com/rust-lang/cargo/issues/2608).
+While this alternative meets the popular aesthetic preference of having generic
+commands with a large variety of usecases (over single-purpose commands), using
+yanking this way has a few drawbacks:
+
+- Cargo doesn't allow yanked packages to be used in new packages. In the
+  author's opinion, people who know what they're doing should be allowed to
+  depend on vulnerable packages, as they might use the crate in a way that
+  poses no security threat.
+
+  Some vulnerabilities can be mitigated in ways other than upgrading a crate,
+  like making local configuration changes. Some vulnerabilities may affect
+  optional functionality not everyone is using, or functionality that can be
+  compiled out by e.g. disabling certain cargo feature settings for that crate.
+  Some may be relatively innocuous and/or hard-to-exploit and therefore not
+  warrant an immediate upgrade. Sometimes no action (other than setting
+  ``allow_vulnerable = true``) is required at all because the dependent crate
+  never used the vulnerable functionality to begin with.
+
+  At the same time it doesn't make sense to depend on packages that don't
+  compile, and currently yanking is primarily used to mark such packages.
+
+- Cargo doesn't give any advice about further procedure when yanking a package.
+  I think in the context of security vulnerabilities this is very much needed,
+  as few OSS maintainers are exposed to this problem regularly enough to know
+  what they're doing.
+
+- Lastly, the data exposed via the Crates.io would be a lot less structured.
+  Automatic security notifications via third-party tooling would be impossible
+  because there is no way to determine whether a package was yanked because of
+  a security vulnerability or not.
+
+Most of these problems can be fixed by asking the user to attach a "reason" to
+their yanked packages, such as ``security``, ``deprecation``, ``broken`` (and
+then make Cargo's behavior dependent on that). However, at that that point
+``yank`` is no longer generic (as in a function having type parameters), but
+simply a lot of single-purpose commands stuffed into one (as in function
+overloading in Java).  And the name "yank" wouldn't make sense for crate
+versions that may still be available (depending on the "reason").
+
+# Unresolved questions
+[unresolved]: #unresolved-questions
+
+## DWF vs CVE
+
+- It may be counterintuitive that one can specify CVEs in the DWF parameter.
+  Should it be called ``cve`` instead even though it can also be used for DWFs?
```

Comment:
That would be overly complex IMO. IDs should be unique without considering the subkey.

If one one advisory has:
```toml
ids = { foo = "DEADBEEF" }
```
and another one has:
```toml
ids = { bar = "DEADBEEF" }
```
then, what would the following mean:
```toml
allow_vulnerable = ["DEADBEEF"]
```

Or do you also want the subkeys there like:
```toml
allow_vulnerable = [{foo = "DEADBEEF"}]
```
What's the advantage over a simple convention how to format the string?